### PR TITLE
[DNM]executor: use UTC to decode timestamp in index read executor

### PR DESCRIPTION
**Pull request by tiancaiamao**
fix issue https://github.com/pingcap/tidb/issues/3467

### CODE REVIEW

File: executor/executor_test.go [hunk 1]
```diff
@@ -1469,6 +1469,25 @@ func (s *testSuite) TestTimestampTimeZone(c *C) {
 		tk.MustExec(fmt.Sprintf("set time_zone = '%s'", tt.timezone))
 		tk.MustQuery(fmt.Sprintf("select * from t where ts = '%s'", tt.expect)).Check(testkit.Rows(tt.expect))
 	}
+
+	// For issue https://github.com/pingcap/tidb/issues/3467
```
Comment (shenli) [inline executor/executor_test.go]: XSelectIndexExec is not covered.

Comment (tiancaiamao) [inline executor/executor_test.go]: It's covered in tidb-test https://github.com/pingcap/tidb-test/pull/315

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
@coocood @XuHuaiyu PTAL

**Comment by coocood** [comment]
@tiancaiamao 
I think the title should be `use session timezone instead of UTC`.
And why change `new_distsql.go`?

**Comment by tiancaiamao** [comment]
The fix is replaced by this PR https://github.com/pingcap/tidb/pull/3497

**Pull request closed without merging**